src/features/unified-share-modal/ContactsField.js

```diff
@@ -127,6 +127,19 @@ class ContactsField extends React.Component<Props, State> {
 
     debouncedGetContacts = debounce(this.getContactsPromise, 200);
 
+    handleParseItems = (inputValue: string): Array<string> => {
+        const { validator } = this.props;
+
+        // ContactField allows invalid pills to be displayed in
+        // in some cases (e.g., when user is external and external
+        // collab is restricted). We don't allow, however, invalid
+        // emails from the pill selector input to be turned into pills.
+        const emails = parseEmails(inputValue);
+        const validEmails = emails.filter(email => validator(email));
```

Comment:
I wasn't sure where to put this since the parsing seems to be the concern of `ContactsField`, but email validation is mostly handled on `EmailForm`. I thought calling the `validator` provided by `EmailForm` would be a good compromise, but let me know if you have any suggestions.

Reply:
Also not sure if this qualifies as a breaking change because, even though this does not change the behavior of USM itself, it does slightly change the behavior of this component when used on its own (not sure if we currently have use cases for that). Happy to mark the commit as "breaking" if needed.

Reply:
Sub-component APIs in `features/*/**` aren't considered public